src/metric/regression_metric.hpp

```diff
@@ -171,6 +171,19 @@ class L1Metric: public RegressionMetric<L1Metric> {
   }
 };
 
+/*! \brief MAPE loss for regression task */
+class MAPEMetric: public RegressionMetric<MAPEMetric> {
+public:
+  explicit MAPEMetric(const MetricConfig& config) :RegressionMetric<MAPEMetric>(config) {}
+
+  inline static double LossOnPoint(label_t label, double score, const MetricConfig&) {
+    return 100 * std::fabs(score - label) / label;
```

Comment:
label could be zero.

Reply:
What do you suggest in such a case?

Reply:
maybe, when label=0 and score=0, return 0, 
otherwise return score directly.

Reply:
@Laurae2 any suggestions  ?

Reply:
Do you mean

```
if (label == 0 && score == 0) => return 0;
if (label == 0 && score != 0) => return score;
otherwise return (score - label) / label;
```